Library/Homebrew/extend/os/linux/software_spec.rb

```diff
@@ -1,6 +1,40 @@
 # typed: strict
 # frozen_string_literal: true
 
+class SoftwareSpec
+  extend T::Sig
+
+  GNU_GCC_REGEXP = T.let(/^(gcc)$|^(gcc@(4\.9|[5-9]|[10-99]))$/.freeze, Regexp)
+
+  sig { params(spec: T.any(String, T::Hash[T.untyped, T.untyped])).void }
+  def remove_gnu_compilers(spec)
+    spec = spec.keys.first if spec.is_a?(Hash)
+    return unless spec.is_a?(String)
+    return unless spec.match?(GNU_GCC_REGEXP)
+
+    # When a gcc compiler is defined, remove the other
```

Comment:
What does it mean for a compiler to be "defined"? How does one do that?

Reply:
I updated the text to be more clear (I hope).